src/App.tsx

```diff
@@ -1,98 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+    .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
-  return (
-    <div className="container">
-      <h1>Items with Pagination</h1>
+    const total = items.length;
+    const [itemsPerPage, setItemsPerPage] = useState(5);
+    const [currentPage, setCurrentPage] = useState(1);
 
-      <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
-      </p>
+    const firstItem = currentPage * itemsPerPage - itemsPerPage + 1;
+    const lastItem = currentPage * itemsPerPage > total
+        ? total
+        : currentPage * itemsPerPage;
 
-      <div className="form-group row">
-        <div className="col-3 col-sm-2 col-xl-1">
-          <select
-            data-cy="perPageSelector"
-            id="perPageSelector"
-            className="form-control"
-          >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
-          </select>
-        </div>
+    const visibleItems = items.slice(firstItem - 1, lastItem);
+
+    function perPageChange(event: React.ChangeEvent<HTMLSelectElement>) {
+        setItemsPerPage(+event.target.value);
+        setCurrentPage(1);
+    }
+    return (
+        <div className="container">
+            <h1>Items with Pagination</h1>
+
+            <p className="lead" data-cy="info">
+                Page {currentPage} (items {firstItem} - {lastItem} of {total})
+            </p>
+
+            <div className="form-group row">
+                <div className="col-3 col-sm-2 col-xl-1">
+                    <select
+                        data-cy="perPageSelector"
+                        id="perPageSelector"
+                        className="form-control"
+                        onChange={(event) => { perPageChange(event) }}
```

Comment:
pls follow [naming convention](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) when giving names to functions